integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java

```diff
@@ -0,0 +1,131 @@
+    private static final int COLLECTOR_PORT = 14250;
+    private static final int HEALTH_PORT = 14269;
+    private static final String JAEGER_URL = "http://localhost";
+    @Inject
+    OpenTelemetry openTelemetry;
+    private static final DockerClient dockerClient;
+
+    static {
+        DockerClientConfig config = DefaultDockerClientConfig.createDefaultConfigBuilder().build();
+        DockerHttpClient httpClient = new ZerodepDockerHttpClient.Builder()
+                .dockerHost(config.getDockerHost())
+                .sslConfig(config.getSSLConfig())
+                .build();
+        dockerClient = DockerClientImpl.getInstance(config, httpClient);
+        if (dockerClient.listContainersCmd().exec().stream()
+                .noneMatch(container -> container.getNames()[0].equals("/jaeger"))) {
+            ExposedPort queryPort = ExposedPort.tcp(QUERY_PORT);
+            ExposedPort collectorPort = ExposedPort.tcp(COLLECTOR_PORT);
+            ExposedPort hostPort = ExposedPort.tcp(HEALTH_PORT);
+            Ports portBindings = new Ports();
+            portBindings.bind(queryPort, Ports.Binding.bindPort(QUERY_PORT));
+            portBindings.bind(collectorPort, Ports.Binding.bindPort(COLLECTOR_PORT));
+            portBindings.bind(hostPort, Ports.Binding.bindPort(HEALTH_PORT));
+            CreateContainerResponse container = dockerClient
+                    .createContainerCmd("ghcr.io/open-telemetry/opentelemetry-java/jaeger:1.32")
+                    .withExposedPorts(queryPort, collectorPort, hostPort)
+                    .withHostConfig(newHostConfig().withPortBindings(portBindings))
+                    .withName("jaeger")
+                    .exec();
+            dockerClient.startContainerCmd(container.getId()).exec();
+        }
+    }
+
+    @AfterAll
+    static void teardown() {
+        dockerClient.listContainersCmd().exec()
+                .forEach(container -> {
+                    dockerClient.stopContainerCmd(container.getId()).exec();
+                    dockerClient.removeContainerCmd(container.getId()).exec();
+                });
+    }
+
+    @Test
+    void testJaegerRemoteIntegration() {
+        createTestSpan(openTelemetry);
+        Awaitility.await()
+                .atMost(Duration.ofSeconds(30))
+                .until(OTelJaegerRemoteTest::assertJaegerHaveTrace);
+    }
+
+    private void createTestSpan(OpenTelemetry openTelemetry) {
+        Span span = openTelemetry.getTracer(getClass().getCanonicalName()).spanBuilder("Test span").startSpan();
+        try (Scope scope = span.makeCurrent()) {
+            span.addEvent("Test event");
+        } catch (Throwable t) {
+            span.recordException(t);
+            throw t;
+        } finally {
+            span.end();
+        }
+    }
+
+    private static boolean assertJaegerHaveTrace() {
+        try {
+
+            String serviceName = ConfigProvider.getConfig().getConfigValue("quarkus.application.name").getValue();
```

Comment:
Couldn't you just use restasured?
There's many examples on other OTel tests that can make this test more legible and maintainable... No need to be parsing json.